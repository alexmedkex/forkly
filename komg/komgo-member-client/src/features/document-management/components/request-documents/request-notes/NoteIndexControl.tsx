import * as React from 'react'
import styled from 'styled-components'
import { Icon } from 'semantic-ui-react'

export interface Props {
  currentIndex: number
  maxIndex: number
  setIndex(newIndex: number): void
}

export const NoteIndexControl = (props: Props) => {
  const { currentIndex, maxIndex, setIndex } = props
  const displayIndex = currentIndex + 1
  return (
    <InlineFlex>
      <Icon
        name="chevron left"
        size="small"
        disabled={displayIndex === 1}
        onClick={e => (displayIndex > 1 ? setIndex(currentIndex - 1) : void 0)}
        data-test-id="incoming-note-btn-left"
      />
      <CurrentIndex data-test-id="incoming-note-current-index">{`${displayIndex}/${maxIndex}`}</CurrentIndex>
      <Icon
        name="chevron right"
        size="small"
        disabled={displayIndex === maxIndex}
        onClick={e => (displayIndex < maxIndex ? props.setIndex(currentIndex + 1) : void 0)}
        data-test-id="incoming-note-btn-right"
      />
    </InlineFlex>
  )
}

const InlineFlex = styled.div`
  display: inline-flex;
  align-items: center;
`

const CurrentIndex = styled.div`
  height: 21px;
  width: 33px;
  color: #5d768f;
  font-size: 11px;
  line-height: 21px;
  text-align: center;
`
