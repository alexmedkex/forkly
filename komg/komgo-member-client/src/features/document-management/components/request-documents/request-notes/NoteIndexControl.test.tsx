import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { NoteIndexControl, Props } from './NoteIndexControl'

describe('NoteIndexControl', () => {
  it('should match default snapshot', () => {
    const props: Props = {
      currentIndex: 0,
      maxIndex: 1,
      setIndex: jest.fn((toIndex: number) => void 0)
    }
    expect(renderer.create(<NoteIndexControl {...props} />).toJSON()).toMatchSnapshot()
  })
})
