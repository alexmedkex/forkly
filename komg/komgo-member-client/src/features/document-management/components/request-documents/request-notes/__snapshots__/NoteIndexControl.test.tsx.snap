// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteIndexControl should match default snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  height: 21px;
  width: 33px;
  color: #5d768f;
  font-size: 11px;
  line-height: 21px;
  text-align: center;
}

<div
  className="c0"
>
  <i
    aria-hidden="true"
    className="chevron left small disabled icon"
    data-test-id="incoming-note-btn-left"
    onClick={[Function]}
  />
  <div
    className="c1"
    data-test-id="incoming-note-current-index"
  >
    1/1
  </div>
  <i
    aria-hidden="true"
    className="chevron right small disabled icon"
    data-test-id="incoming-note-btn-right"
    onClick={[Function]}
  />
</div>
`;
